battery_check.c

```diff
@@ -0,0 +1,105 @@
+#include <stdio.h>
+#include "battery_check.h"
+
+#define MAX_BATTERY_VOLTAGE (4.3)
+#define MIN_BATTERY_VOLTAGE (2.3)
+
+
+// REFERENCE : https://www.electronics-notes.com/articles/electronic_components/battery-technology/li-ion-lithium-ion-charging.php#:~:text=Constant%20current%20charge:%20In%20the%20first%20stage%20of,rate%20of%20a%20maximum%20of%200.8C%20is%20recommended.
+
+
+static STRUCT_BATTERY_MANUFACTURER_LIMITS manufacturer_one;
+static STRUCT_BATTERY_MANUFACTURER_LIMITS manufacturer_two;
+
+int battery_check_temperature_ok(int temperature)
+{
+	int ret_val = 1;
+	
+	if((temperature < manufacturer_one.batt_temp_limits.min_oper_temperature)  || 
+	   (temperature > manufacturer_one.batt_temp_limits.max_oper_temperature)) 
+	{
+    	printf("Temperature out of range!\n");
```

Comment:
printf() statement duplicated in all the function . separate impure code from pure code